poc/idpf.sage

```diff
@@ -0,0 +1,57 @@
+# Definition of IDPFs.
+
+from __future__ import annotations
+from sagelib.common import Unsigned
+import sagelib.field as field
+import sagelib.prg as prg
+
+# An Incremntal Distributed Point Function (IDPF).
+class Idpf:
+    # Number of keys gennerated by the IPDF-key generation algorithm.
+    SHARES: Unsigned = None
+
+    # Bit length of valid input values (i.e., values of `alpha`).
+    BITS: Unsigned = None
+
+    # The finite field used to reprresent the inner nodes of the IDPF tree.
+    FieldInner: field.Field = None
+
+    # The finite field used to represent the leaf nodes of the IDPF tree.
+    FieldLeaf: field.Field = None
```

Comment:
@schoppmp we had previously talked about allowing each level of the hierarchy to have its own field. I decided to just define two fields, one for the leaves and another for the inner nodes. That's sufficient to admit the construction on the Poplar paper. Do you think more flexibility will be desirable?

Removing this flexibility seems useful for keeping things simple.